include/fmt/ranges.h

```diff
@@ -660,6 +660,43 @@ struct formatter<tuple_join_view<Char, T...>, Char> {
   }
 };
 
+namespace detail {
+// Check if T has an interface like container adapter (e.g. std::stack,
+// std::queue, std::priority_queue).
+template <typename T> class is_container_adaptor_like {
+  template <typename U> static auto check(U* p) -> typename U::container_type;
+  template <typename> static void check(...);
+
+ public:
+  static constexpr const bool value =
+      !std::is_void<decltype(check<T>(nullptr))>::value;
+};
+}  // namespace detail
+
+template <typename T, typename Char>
+struct formatter<T, Char,
+                 enable_if_t<detail::is_container_adaptor_like<T>::value>> {
+ private:
+    struct formatter<typename T::container_type, Char> container_formatter;
```

Comment:
`struct` not needed here and I suggest using inheritance instead of composition to avoid redefining `parse`.

Also please apply clang-format, indent is a bit off.